userscript/lovely-forks.user.js

```diff
@@ -0,0 +1,501 @@
+
+    var iconPath = document.createElementNS(svgNS, 'path');
+    switch(type) {
+        case 'star':
+            title.appendChild(document.createTextNode('Number of stars'));
+            iconPath.setAttributeNS(null, 'd', 'M14 6l-4.9-0.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14l4.33-2.33 4.33 2.33L10.4 9.26 14 6z');
+            break;
+        case 'flame':
+            title.appendChild(document.createTextNode('Fork may be more recent than upstream.'));
+            iconPath.setAttributeNS(null, 'd',  'M5.05 0.31c0.81 2.17 0.41 3.38-0.52 4.31-0.98 1.05-2.55 1.83-3.63 3.36-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-0.3-6.61-0.61 2.03 0.53 3.33 1.94 2.86 1.39-0.47 2.3 0.53 2.27 1.67-0.02 0.78-0.31 1.44-1.13 1.81 3.42-0.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52 0.13-2.03 1.13-1.89 2.75 0.09 1.08-1.02 1.8-1.86 1.33-0.67-0.41-0.66-1.19-0.06-1.78 1.25-1.23 1.75-4.09-1.88-6.22l-0.02-0.02z');
+            iconPath.setAttributeNS(null, 'fill', '#d26911');
+            break;
+    }
+
+    iconPath.appendChild(title);
+    svg.appendChild(iconPath);
+
+    return svg;
+}
+
+function emptyElem(elem) {
+    elem.textContent = ''; // How jQuery does it
+}
+
+function mbStrToMs(dateStr) {
+    return dateStr !== null ? Date.parse(dateStr) : null;
+}
+
+function isExpired(timeMs) {
+    var currentTime = new Date();
+
+    // The time of expiry of data is set to be an hour ago
+    var expiryTimeMs = currentTime.valueOf() - 1000 * 60 * 60;
+    return timeMs < expiryTimeMs;
+}
+
+function makeSelfDataKey(user, repo) {
+    return 'lovely-forks@self:' + user + '/' + repo;
+}
+
+function makeRemoteDataKey(user, repo) {
+    return 'lovely-forks@remote:' + user + '/' + repo;
+}
+
+
+var reDateKey = new RegExp('^lovely-forks@date:(.*)/(.*)$');
+function makeTimeKey(user, repo) {
+    return 'lovely-forks@date:' + user + '/' + repo;
+}
+
+function parseTimeKey(key) {
+    var match = reDateKey.exec(key);
+    if (match !== null) {
+        return [match[1], match[2]];
+    } else {
+        return null;
+    }
+}
+
+function getForksElement() {
+    // Verify that the element exists and it's still valid
+    // otherwise, create it
+    if (document.body.contains(text)) {
+        return text;
+    }
+
+    // If the layout of the page changes, we'll have to change this location.
+    // We should make sure that we do not accidentally cause errors here.
+    var repoName = document.querySelector('.repohead-details-container .public');
+    if (repoName) {
+        try {
+            text = document.createElement('span');
+
+            // Stealing the styling from GitHub fork-info
+            text.classList.add('fork-flag', 'lovely-forks-addon');
+
+            repoName.appendChild(text);
+
+            return text;
+        } catch (e) {
+            GM_log(_logName,
+                          'Error appending data to DOM',
+                          e);
+        }
+    } else {
+        GM_log(_logName,
```

Comment:
This may be my ignorance, but I'm assuming that `GM` stands for GreaseMonkey?

I'm testing it with Tampermonkey and at least editor says that it couldn't find `GM_log`. Is there a reliable way to shim logging across user-script engines?